src/components/RewardCard.tsx

```diff
@@ -0,0 +1,31 @@
+import { Button, Card, Chip, Box, Typography } from "@mui/material"
+import { useState } from "react";
+
+const RewardCard = ({ label, status, reward }: React.PropsWithChildren<{ label: String, status: String, reward: boolean }>) => {
+
+    const [redeem, set_redeem] = useState("Redeem");
+
+    const handleClick = () => {
+        set_redeem("Redeemed");
```

Comment:
Let's pass the state upwards to the parent.

I want to be able to do something like:

```ts
const MyComponent = () => {
    const some_thing = () => {
         alert("Hello world");
    }

    return (
        <RewardCard on_redeem={some_thing} reward={true} />
    );
}

``